lib/models/entitiesStore.js

```diff
@@ -28,15 +28,89 @@ var async = require('async'),
     entitiesCollectionName = require('../../config').orionDb.collection,
     myutils = require('../myutils'),
     constants = require('../constants'),
-    logger = require('logops');
+    logger = require('logops'),
+    NGSI = require('ngsijs');
 
 function orionServiceDb(service) {
     return appContext.OrionDb(config.orionDb.prefix + '-' + service);
 }
 
-function findSilentEntities(service, subservice, ruleData, func, callback) {
+function findSilentEntitiesByAPI(service, subservice, ruleData, func, callback) {
+    var context = { op: 'checkNoSignal.findSilentEntitiesByAPI', comp: constants.COMPONENT_NAME };
+
+    var options = {
+        service: service,
+        servicepath: subservice
+    };
+    options.headers = {};
+    // if (token !== null) {
+    //     options.headers[constants.AUTH_HEADER] = token;
+    // }
+    // Add correlator
+    var domain = process.domain;
+    if (domain && domain.context) {
+        options.headers[constants.CORRELATOR_HEADER] = domain.context.corr;
+        // Add other headers
+        if (domain.context.srv && options.headers[constants.SERVICE_HEADER] === undefined) {
+            options.headers[constants.SERVICE_HEADER] = domain.context.srv;
+        }
+        if (domain.context.subsrv && options.headers[constants.SUBSERVICE_HEADER] === undefined) {
+            options.headers[constants.SUBSERVICE_HEADER] = domain.context.subsrv;
+        }
+        if (domain.context.from && options.headers[constants.REALIP_HEADER] === undefined) {
+            options.headers[constants.REALIP_HEADER] = domain.context.from;
+        }
+    }
+    var connection = new NGSI.Connection(config.orion.URL, options);
+
+    var filter = {
+        service: service,
+        servicepath: subservice,
+        type: ruleData.type,
+        mq: ruleData.attribute + '.dateModified<' + (Date.now() / 1000 - ruleData.reportInterval).toString()
+    };
+    if (ruleData.id) {
+        filter.id = ruleData.id;
+    } else if (ruleData.idRegexp) {
+        filter.idPattern = ruleData.idRegexp;
+    }
+    logger.debug(
+        context,
+        'find silent entities by API ngsi using options %j and filter %j and rule %j',
+        options,
+        filter,
+        ruleData
+    );
+    // https://ficodes.github.io/ngsijs/stable/NGSI.Connection.html#.%22v2.listEntities%22__anchor
+    connection.v2.listEntities(filter).then(
```

Comment:
the library takes pagination into account? it's there a problem with thousand of entities? I think it does, but check.

Reply:
Solved in these commits: https://github.com/telefonicaid/perseo-fe/pull/726/commits/aa79f8612ce24b6577226b69ae7d464ddd797a51#diff-d21abada9564b2a8fdaf4c061d51c73f386a5b4869659ab7517d94c9f665818c , https://github.com/telefonicaid/perseo-fe/pull/726/commits/cb70ed8808458d986be3ac2119e1eb90a7b67adc